WooCommerce/WooCommerceScreenshots/WooCommerceScreenshots.swift

```diff
@@ -54,6 +60,67 @@ class WooCommerceScreenshots: XCTestCase {
             .selectProduct(atIndex: 1)
             .thenTakeScreenshot(named: "product-details")
     }
+
+    private let port = Int.random(in: 2048...65535)
+
+    private let loop = try! SelectorEventLoop(selector: try! KqueueSelector())
+
+    private let queue = DispatchQueue(label: "screenshots-asset-server")
+
+    private lazy var server = DefaultHTTPServer(eventLoop: loop, port: port, app: handleWebRequest)
+
+    func startWebServer() {
+        queue.async {
+            /// Start HTTP server to listen on the port
+            try! self.server.start()
+
+            /// Run event loop
+            self.loop.runForever()
+
+            debugPrint("Web Server running on port \(self.port)")
+        }
+    }
+
+    func handleWebRequest(env: [String: Any], startResponse: ((String, [(String, String)]) -> ()), sendBody: ((Data) -> ())) {
+        /// Extract the path, so http://localhost:9285/foo-bar would be `/foo-bar` here.
+        guard let path = env["PATH_INFO"] as? String else {
+            startResponse("404 Not Found", [])
+            sendBody(Data())
+            return
+        }
+
+        /// Remove the leading `/`, so we're left with `foo-bar`
+        let assetName = path.replacingOccurrences(of: "/", with: "")
```

Comment:
This feels a bit dangerous, especially if we start using namespaced assets at some point, e.g.
<img width="1140" alt="namespaced-xcasset" src="https://user-images.githubusercontent.com/216089/103890556-8f323880-50e8-11eb-9e08-b08e8fbc4afc.png">
(which then are referenced in the code by `UIImage(named: "icons/icon_add")` using their fully qualified name)

Maybe this could be safer instead:
```suggestion
        let assetName = path.replacingOccurrences(of: "^/", with: "", options: .regularExpression)
```

Reply:
Ah interesting ok!

Sounds good to me!